lib/storage/src/content_manager/consensus_state.rs

```diff
@@ -917,4 +931,47 @@ mod tests {
         // Even when `max_size` is `0` this fn should return at least 1 entry
         assert_eq!(wal.entries(4, 5, Some(0)).unwrap().len(), 1)
     }
+
+    struct NoCollections;
+
+    impl CollectionContainer for NoCollections {
+        fn perform_collection_meta_op(
+            &self,
+            _operation: crate::content_manager::collection_meta_ops::CollectionMetaOperations,
+        ) -> Result<bool, crate::content_manager::errors::StorageError> {
+            Ok(true)
+        }
+
+        fn collections_snapshot(&self) -> super::CollectionsSnapshot {
+            super::CollectionsSnapshot::default()
+        }
+
+        fn apply_collections_snapshot(
+            &self,
+            _data: super::CollectionsSnapshot,
+        ) -> Result<(), crate::content_manager::errors::StorageError> {
+            Ok(())
+        }
+    }
+
+    prop_compose! {
+        fn gen_entries(max_entries: u64)(n in 0u64..max_entries) -> Vec<Entry> {
+            (1..(n+1)).into_iter().map(|index| Entry {index, term: 1, ..Default::default()}).collect::<Vec<Entry>>()
```

Comment:
Is it possible to increment the term from time to time as well?
Could be using a randomized approach or just modding around the index.